### ISSUE TITLE
[Editable] onSubmit of custom controls not saving new value

### ISSUE THREAD

**Issue by jana-uoft** [chakra-ui/chakra-ui#630]
**Describe the bug**
Editable component with custom controls does not work as intended. 
This can be seen in the site docs as well. 
https://chakra-ui.com/editable#with-custom-controls

**To Reproduce**
Click the edit icon and update the value; and when clicking the `check` button, the new value does not persist. However, it works when just pressing the `enter` key.

**Comment by jana-uoft** [comment]
Digging around, I believe the error is coming from the input losing focus on clicking the `check` button and resetting its value. 

I was able to fix this by setting `submitOnBlur` to the `Editable` component. 
```
                <Editable
                  defaultValue={media.title}
                  onSubmit={(title) => updateMedia({ variables: { id: media.id, data: { title } } })}
                  isPreviewFocusable={false}
                  submitOnBlur
                  display='flex'
                  justifyContent='space-between'
                >
                 ......
```

Side note: Having `isPreviewFocusable` set to true causes tabbing to get stuck inside the input component. So I set that as false.

**Comment by tafelito** [comment]
Shouldn't EditableInput also use the Input instead of a PseudoBox in order keep consistency?

**Comment by with-heart** [comment]
This has been fixed in 1.0: https://codesandbox.io/s/friendly-williamson-1oslz?file=/src/App.js
